pkg/registry/apis/dashboard/legacy/sql_dashboards.go

```diff
@@ -428,6 +428,11 @@ func (a *dashboardSqlAccess) buildSaveDashboardCommand(ctx context.Context, orgI
 		return nil, created, err
 	}
 
+	// Run the conversion at read time
+	if dash.Status.Conversion.Failed {
+		apiVersion = "v0alpha1"
```

Comment:
@stephaniehingtgen I added this for legacy -- when a conversion fails it was written to v1 -- we can save as v0 and the error will be run again (and added to the output) on read.

With unified storage, the error is saved in the output so that is good enough